lib/std/debug.zig

```diff
@@ -1781,8 +1782,8 @@ fn handleSegfaultLinux(sig: i32, info: *const os.siginfo_t, ctx_ptr: ?*const c_v
         },
         .x86_64 => {
             const ctx = @ptrCast(*const os.ucontext_t, @alignCast(@alignOf(os.ucontext_t), ctx_ptr));
-            const ip = @intCast(usize, ctx.mcontext.gregs[os.REG_RIP]);
-            const bp = @intCast(usize, ctx.mcontext.gregs[os.REG_RBP]);
+            const ip = @intCast(usize, if (builtin.os.tag == .freebsd) ctx.mcontext.rip else ctx.mcontext.gregs[os.REG_RIP]);
```

Comment:
could you use a `switch (builtin.os.tag)` for `ip` and for `bp` ? it would simplify the merge with #7487 (where OpenBSD will need special case too)

Reply:
Sure! Thanks for the heads up, I didn't see that PR.

Reply:
Ok, done.